src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -1310,16 +1312,18 @@ public function searchProductsAction(Request $request): Response
     {
         try {
             $searchPhrase = $request->query->get('search_phrase');
+            $currencyId = $request->query->get('search_currency');
+            $currencyIsoCode = $currencyId !== null
+                ? Currency::getIsoCodeById((int) $currencyId)
+                : Currency::getIsoCodeById((int) Configuration::get('PS_CURRENCY_DEFAULT'));
 
             /** @var FoundProduct[] $foundProducts */
-            $foundProducts = $this->getQueryBus()->handle(new SearchProducts($searchPhrase, 10));
+            $foundProducts = $this->getQueryBus()->handle(new SearchProducts($searchPhrase, 10, $currencyIsoCode));
 
-            $serializer = $this->get('prestashop.bundle.snake_case_serializer_json');
-
-            return new Response($serializer->serialize($foundProducts, 'json'));
+            return $this->json($foundProducts);
```

Comment:
I just noticed that in another PR the format of this API is a bit different https://github.com/PrestaShop/PrestaShop/pull/16207/files#diff-16fa3813bd595f723846906e6d01cfefR1318

It uses:
```php
      return $this->json([
                'products' => $foundProducts,
            ]);
```

So we need to make both PRs match on a common format, but I thought adding this extra `products` field makes sens in case the API needs to return additional data (with your format it will be more complicated) What do you think?

Reply:
:ballot_box_with_check: 
**Bonus :** it modifies one another file admin-dev/themes/new-theme/js/pages/order/create/product-manager.js